test/system/fidelity_test_03.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+RED='\e[41m'
+GREEN='\e[42m'
+NC='\033[0m' # No Color
+BOLD='\033[1m'
+UNDERLINE='\033[4m'
+
+print_header() {
+    echo
+    echo "**************************************************"
+    echo "* Starting Fidelity Test 03 with path $directory_path"
+    echo "**************************************************"
+    echo "Test 03: Checks if the events have been written in sequential order."
+    echo
+}
+
+# Check if the correct number of arguments are provided
+if [ "$#" -ne 2 ]; then
+    echo "Usage: $0 <input_file> <directory>"
+    exit 1
+fi
+
+verify_order() {
+    local file1="$1"
+    local file2="$2"
+
+    # Check if both files are specified
+    if [[ -z "$file1" || -z "$file2" ]]; then
+        echo "Error: Two file paths must be provided"
+        return 1
+    fi
+
+    # Read file1 line by line
+    local current_line_number=0
+    local found
+    local last_line_number=0
+
+
+    while IFS= read -r line_from_file1; do
```

Comment:
Is it possible to avoid this two level nested while loop? For example, check ordering in each file first. Then merge sort all generated files and compare the merged file with the original one. With this, we can reduce the complexity to O(nlogn).

Reply:
Based on what I understood we want to test each keeper correctness independently and therefore this needs to be checked on individual files. As I cannot merge the same storyId files, I need to check each of them independently and for checking the order in that scenario I need to assume there are events that will be missing. For checking it in this situation I need to keep the position of both files and have a two level nested while loop